src/index.js

```diff
@@ -6,51 +6,85 @@ const tags = tagsW3.map(tag => tag.replace(/[<|>]/g, ""));
+    form.addEventListener("submit", function (event) {
+        // on submit
+        event.preventDefault(); // Prevent the default form submission
+        submitVal(); // call our submitVal function
+    });
+
+    function submitVal() {
+        let val = document.getElementById("input").value;
+        // remove all white spaces and non-alphabetical characters except . ! - 
+        let regex = /[^a-zA-Z!-.]/g;
+        // change input to lower case and apply regex
+        let sanitizedVal = val.toLowerCase().replace(regex, "");
+        let li = "<li>" + sanitizedVal + "</li>";
+        // check if the input value is in the tags array
+        let found = tags.some(tag => tag.toLowerCase() === sanitizedVal);
+        // convert the list to an array
+        const getList = JSON.parse(localStorage.getItem("myTags"));
+        let checkDuplicate = false;
+        // ? checks if the list is not empty first, to make includes() work
+        if (getList?.includes(sanitizedVal)) {
+        checkDuplicate = true;
+        }
+        if (found && !checkDuplicate) {
+            // add tag to the exisitng array of tags 
+            let updatedTags = insertTagAndSort(sanitizedVal, savedTags);
+            // save the updated array of tags to localStorage
+            localStorage.setItem("myTags", JSON.stringify(updatedTags));
+            // render the updated list of tags
+            renderAllTags(updatedTags);
+            updateProgressBar();
+        }
+        document.getElementById("input").value = "";
     }
-    document.getElementById("input").value = "";
-  }
 });
 
-function saveTagToLocalStorage(tag) {
-  let savedTags = JSON.parse(localStorage.getItem("myTags")) || [];
-  savedTags.push(tag);
-  localStorage.setItem("myTags", JSON.stringify(savedTags));
+// deals with the data aspect of adding a tag and sorting the list
+function insertTagAndSort(tag, tagArray) {
+    tagArray.push(tag);
+    tagArray.sort();
+    return tagArray;
+}
+
+// deals with the DOM aspect (ie rendering) of the tags. used at the initial render and on each new tag
+function renderAllTags(tagsArray) {
+    let listElement = document.getElementById('list');
+    
+    // Clear the current list. Necessary for a correct re-rendering of the updated list
+    listElement.innerHTML = ''; 
+    
+    // lines 66, 69 through 76, and 82 deal with adding a separator between tags that start with different letters
+    let previousTagInitial = '';
+
+    tagsArray.forEach(tag => {
+        let currentTagInitial = tag[0];
+        // if the current tag's initial is different from the previous tag's initial, create a space
+        if (previousTagInitial && currentTagInitial !== previousTagInitial) {
+            // create a space. this is a div and not an li to not confuse screen readers
+            let separator = document.createElement("div");
+            separator.style.height = "16px";
+            listElement.appendChild(separator);
```

Comment:
I will do some research on this part. Adding a `div` is not appropriate because a `ul` can only have `li` as a direct child. So we must come up with another solution 🔥 .
See: https://dequeuniversity.com/rules/axe/4.7/list?application=AxeFirefox

Reply:
**Possible approach**
- Instead of `div` use `li`, which makes it not fail the parent/child relation.
- Add the attribute `aria-hidden="true"` to this `li` element (should be supported by all major browsers) - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden
- use `attribute selectors` to style this element - https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

💡 I tested this approach with VoiceOver on MacOS and these list items are skipped by screen reader.

Reply:
I purposefully did not use an `li `so not to confuse the screen reader, but did now know `divs` are not allowed inside a `ul`. 

`aria-hidden` tot he rescue then!

And thank you so much for doing the research on this!

Reply:
Are you ok with this approach? The code dealing with separating the groups it's pretty independent of the rest of the function's code (in the sense that we can easily comment it out) if we want to try a different approach in the future.

Reply:
Yes, I think it's good. Maybe we see other/better approaches somewhere else in the future, then we can improve. For now, I guess it is the best solution when we want to keep a space between each group.